framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -736,6 +726,35 @@ public AnnotationMirror getAnnotationFromJavaExpressionString(
     }
 
     /**
+     * Produces the receiver associated with expression on currentPath.
+     *
+     * @param expression Java expression
+     * @param currentPath location at which expression is evaluated
+     * @return
+     */
+    public FlowExpressions.Receiver getReceiverFromJavaExpressionString(
+            String expression, TreePath currentPath) {
+        TypeMirror enclosingClass = InternalUtils.typeOf(TreeUtils.enclosingClass(currentPath));
+
+        FlowExpressions.Receiver r =
+                FlowExpressions.internalRepOfPseudoReceiver(currentPath, enclosingClass);
+        FlowExpressionParseUtil.FlowExpressionContext context =
+                new FlowExpressionParseUtil.FlowExpressionContext(
+                        r,
+                        FlowExpressions.getParametersOfEnclosingMethod(this, currentPath),
+                        this.getContext());
+
+        FlowExpressions.Receiver receiver;
+        try {
+            receiver = FlowExpressionParseUtil.parse(expression, context, currentPath, true);
+        } catch (FlowExpressionParseUtil.FlowExpressionParseException ex) {
+            // ignore parse errors.
+            return null;
```

Comment:
I think that that the exception should be handled by the caller, which is why the signature of `getAnnotationFromJavaExpressionString` says it throws the error.  FlowExpressoinParseExceptions contain a message about why the parsing failed.  Calls of this method might want to surface that information to the user.